packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -315,13 +337,22 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
                         )
                       : this.renderTriggerButton(styles, rtl, getToggleButtonProps)}
                   </div>
-                  {Indicator.create(toggleIndicator, {
-                    defaultProps: {
-                      direction: isOpen ? 'top' : 'bottom',
-                      onClick: getToggleButtonProps().onClick,
-                      styles: styles.toggleIndicator,
-                    },
-                  })}
+                  {showClearIndicator
+                    ? Icon.create(clearIndicator, {
+                        defaultProps: {
+                          className: Dropdown.slotClassNames.clearIndicator,
+                          onClick: this.handleClear,
```

Comment:
Shouldn't onClick be on the override props? Then you can do all necessary for the Dropdown, and then invoke the user's onClick if provided.

Reply:
Good catch 👍